layers.go

```diff
@@ -72,6 +73,28 @@ type Layer struct {
 	// is set before using it.
 	Created time.Time `json:"created,omitempty"`
 
+	// CompressedDigest is the digest of the blob that was last passed to
+	// ApplyDiff() or Put(), as it was presented to us.
+	CompressedDigest digest.Digest `json:"compressed-diff-digest,omitempty"`
+
+	// CompressedSize is the length of the blob that was last passed to
```

Comment:
This'll be useful in the image library.  It'll be able to easily determine if there's some layer already in local storage that was initialized using a compressed diff that had this digest.